internal/handlers/commonparams/params.go

```diff
@@ -74,59 +74,134 @@ func GetWholeNumberParam(value any) (int64, error) {
 		}
+	}
 
-		return whole, nil
+	if whole < 0 {
+		return 0, commonerrors.NewCommandError(
+			commonerrors.ErrValueNegative,
+			fmt.Errorf("BSON field '%s' value must be >= 0, actual value '%d'", param, whole),
+		)
 	}
 
-	switch {
-	case errors.Is(err, ErrUnexpectedType):
-		if _, ok := value.(types.NullType); ok {
-			return 0, nil
+	return whole, nil
+}
+
+// getOptionalPositiveNumber returns doc's value for key or protocol error for invalid parameter.
+func getOptionalPositiveNumber(key string, value any) (int64, error) {
+	whole, err := GetWholeNumberParam(value)
+	if err != nil {
+		switch {
+		case errors.Is(err, ErrUnexpectedType):
+			return 0, commonerrors.NewCommandErrorMsgWithArgument(
+				commonerrors.ErrBadValue,
+				fmt.Sprintf("%s must be a number", key),
+				key,
+			)
+		case errors.Is(err, ErrNotWholeNumber):
+			if _, ok := value.(float64); ok {
+				return 0, commonerrors.NewCommandErrorMsgWithArgument(
+					commonerrors.ErrBadValue,
+					fmt.Sprintf("%v has non-integral value", key),
+					key,
+				)
+			}
+
+			return 0, commonerrors.NewCommandErrorMsgWithArgument(
+				commonerrors.ErrBadValue,
+				fmt.Sprintf("%s must be a whole number", key),
+				key,
+			)
+		default:
+			return 0, lazyerrors.Error(err)
 		}
+	}
 
+	if whole > math.MaxInt32 || whole < math.MinInt32 {
 		return 0, commonerrors.NewCommandErrorMsgWithArgument(
-			commonerrors.ErrTypeMismatch,
-			fmt.Sprintf(
-				`BSON field '%s.%s' is the wrong type '%s', expected types '[long, int, decimal, double]'`,
-				command, param, AliasFromType(value),
-			),
-			param,
+			commonerrors.ErrBadValue,
+			fmt.Sprintf("%v value for %s is out of range", whole, key),
+			key,
 		)
-	case errors.Is(err, ErrNotWholeNumber):
-		if math.Signbit(value.(float64)) {
-			return 0, commonerrors.NewCommandError(
-				commonerrors.ErrValueNegative,
-				fmt.Errorf("BSON field '%s' value must be >= 0, actual value '%d'", param, int(math.Ceil(value.(float64)))),
-			)
-		}
+	}
 
-		// for non-integer numbers, value is rounded to the greatest integer value less than the given value.
-		return int64(math.Floor(value.(float64))), nil
+	if whole < 0 {
+		return 0, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrBadValue,
+			fmt.Sprintf("%v value for %s is out of range", value, key),
+			key,
+		)
+	}
 
-	case errors.Is(err, ErrLongExceededPositive):
-		return math.MaxInt64, nil
+	return whole, nil
+}
 
-	case errors.Is(err, ErrLongExceededNegative):
-		return 0, commonerrors.NewCommandError(
-			commonerrors.ErrValueNegative,
-			fmt.Errorf("BSON field '%s' value must be >= 0, actual value '%d'", param, int(math.Ceil(value.(float64)))),
+// GetBoolOptionalParam returns bool value of v.
+// Non-zero double, long, and int values return true.
+// Zero values for those types, as well as nulls and missing fields, return false.
+// Other types return a protocol error.
+func GetBoolOptionalParam(v any, key string) (bool, error) {
```

Comment:
`getOptionalPositiveNumber(key string, value any)` has the signature key-value, this one has the opposite signature - `value - key`. Let's make them the same, so it's consistent?